File: app/src/main/java/it/rizzoli/atthemoment/model/ListaMezzi.java
package it.rizzoli.atthemoment.model;

import com.google.gson.annotations.SerializedName;
import java.util.List;

public class ListaMezzi {
    @SerializedName("Id")
    private String id;

    @SerializedName("Code")
    private String code;

    @SerializedName("Direction")
    private String direction;

    @SerializedName("Line")
    private Line line;

    @SerializedName("Stops")
    private Stop stops;

    @SerializedName("Geometry")
    private Object geometry;

    @SerializedName("Links")
    private List<Link> links;

    public ListaMezzi() {}

    @Override
    public String toString() {
        return "ListaMezzi{" +
                "id='" + id + '\'' +
                ", code='" + code + '\'' +
                ", direction='" + direction + '\'' +
                ", line=" + line +
                ", stops=" + stops +
                ", geometry=" + geometry +
                ", links=" + links +
                '}';
    }

    public String getId() {
        return id;
    }

    public String getCode() {
        return code;
    }

    public String getDirection() {
        return direction;
    }

    public Line getLine() {
        return line;
    }

    public Stop getStops() {
        return stops;
    }

    public Object getGeometry() {
        return geometry;
    }

    public List<Link> getLinks() {
        return links;
    }
}

File: model/ListaMezzi.swift
import Foundation

struct ListaMezzi: Codable {
    let id: String
    let code: String
    let direction: String
    let line: Line
    let stops: [Stop]?
    let geometry: CodableValue?
    let links: [Link]?

    enum CodingKeys: String, CodingKey {
        case id = "Id"
        case code = "Code"
        case direction = "Direction"
        case line = "Line"
        case stops = "Stops"
        case geometry = "Geometry"
        case links = "Links"
    }
}
